core/src/test/java/org/infinispan/stream/BaseStreamTest.java

```diff
@@ -1489,6 +1489,48 @@ public void testDoubleAllMatch() {
       assertFalse(createStream(entrySet).mapToDouble(toDouble).allMatch(i -> Math.floor(i) == i));
    }
 
+   public void testSkip() {
+      Cache<Double, String> cache = getCache(0);
+      int range = 10;
+      // First populate the cache with a bunch of values
+      IntStream.range(0,range).mapToDouble(value -> value).forEach(i -> cache.put(i, i + "-value"));
+      CacheSet<Map.Entry<Double, String>> entrySet = cache.entrySet();
+
+      int cnt = range;
+      for(int i = 0;i<cnt;i++){
+         List<Map.Entry<Double,String>> res = createStream(entrySet)
+                 .sorted((o1, o2) -> o1.getKey().compareTo(o2.getKey()))
+                 .skip(i).collect(Collectors.toList());
+         for(int j = 0;j< cnt - res.size();j++){
+            String targetValue = (double)j + "-value";
```

Comment:
Ah I see it is because the test passes without it.  Maybe instead of checking to make sure no values exist with those we should be doing the inverse instead.